hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPServiceActor.java

```diff
@@ -486,12 +486,11 @@ DatanodeCommand cacheReport() throws IOException {
 
       cmd = bpNamenode.cacheReport(bpRegistration, bpid, blockIds);
       long sendTime = monotonicNow();
-      long createCost = createTime - startTime;
       long sendCost = sendTime - createTime;
       dn.getMetrics().addCacheReport(sendCost);
       if (LOG.isDebugEnabled()) {
         LOG.debug("CacheReport of " + blockIds.size()
-            + " block(s) took " + createCost + " msecs to generate and "
+            + " block(s) took " + (createTime - startTime) + " msecs to generate and "
```

Comment:
the cost of a subtraction is marginal. Moving that monotonicNow() would be more significant, as if it is using the x86 `RDSTC` call, its a barrier function which blocks all speculative execution until complete

Reply:
IMHO, another reason for this change is that it can improve the maintainability, as the `sendCost` only used by the logging call. But I will leave it alone. We can not move the call to `monotonicNow()`, because the result is also used by  `dn.getMetrics().addCacheReport(sendCost)`.